server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -336,6 +336,9 @@ public NestedLookup nestedLookup() {
     }
 
     public boolean isMultiField(String field) {
+        if (fieldTypeLookup.isRuntimeField(field)) {
+            return false;
+        }
```

Comment:
I see that there are existing callers of this method in FieldMapper. Was it not accurate before? I get thrown off that we look at field mappers later, and runtime fields don't have field mappers, so why do we need this additional if and what effect does it have?

Reply:
This is used when validating `copy_to` fields, in which case we don't really care about runtime fields.  But it is important for field cap filtering, because otherwise we might incorrectly a runtime field `foo.bar` as a multifield of an otherwise existing field `foo`.

Thinking about it though, I reckon we can do this without looking at runtime fields just by checking to see if both the passed-in field name and its parent are field mappers.  Let me see what I can come up with.